tmail-backend/mailbox/encrypted/memory/src/main/java/com/linagora/tmail/encrypted/InMemoryEncryptedEmailContentStore.scala

```diff
@@ -0,0 +1,83 @@
+package com.linagora.tmail.encrypted
+
+import com.google.common.base.Preconditions
+import com.google.inject.{AbstractModule, Scopes}
+import com.linagora.tmail.encrypted.EncryptedEmailContentStore.POSITION_NUMBER_START_AT
+import com.linagora.tmail.encrypted.InMemoryEncryptedEmailContentStore.{emailContentStore, messageIdBlobIdStore}
+import org.apache.james.blob.api.BlobStore.StoragePolicy
+import org.apache.james.blob.api.{BlobId, BlobStore, BucketName}
+import org.apache.james.mailbox.model.MessageId
+import reactor.core.scala.publisher.SMono
+
+import java.nio.charset.StandardCharsets
+import java.util.concurrent.atomic.AtomicInteger
+import javax.inject.Inject
+
+case class EncryptedEmailContentStoreMemoryModule() extends AbstractModule {
+  override def configure(): Unit = {
+    bind(classOf[InMemoryEncryptedEmailContentStore]).in(Scopes.SINGLETON)
+
+    bind(classOf[EncryptedEmailContentStore]).to(classOf[InMemoryEncryptedEmailContentStore])
+  }
+}
+
+object InMemoryEncryptedEmailContentStore {
+  val emailContentStore: scala.collection.concurrent.Map[MessageId, EncryptedEmailContent] = scala.collection.concurrent.TrieMap()
+  val messageIdBlobIdStore: scala.collection.concurrent.Map[MessageId, Map[Int, BlobId]] = scala.collection.concurrent.TrieMap()
```

Comment:
Regarding my above comment then I don't think you need two maps here

Reply:
Like maybe you could create a case class `EncryptedBlobId` with a value String, that would be `encryptedAttachment_{messageId}_{position}`

And you could have just a `Map[EncryptedBlobId, EncryptedEmailContent]`

Reply:
I keep map `Map[Int, BlobId]` (position-blobId) as data structor. And it use to in `def retrieveAttachmentContent(messageId: MessageId, position: Int): BlobId` method. 
If we use  `Map[EncryptedBlobId, EncryptedEmailContent]`, we need more POJO object and maybe duplicate `value` for many `key`